### PR TITLE
Improve Repository Consistency Check in Tests

### PR DESCRIPTION
**Pull request by original-brownbear**
* Check that index metadata as well as snapshot metadata always exists
when referenced by other metadata
   * This is kind of nice to have with all the indices cleanup we're doing since it makes sure we never deleted an index accidentally

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java [hunk 1]
```diff
@@ -113,14 +116,32 @@ private static void assertIndexUUIDs(BlobContainer repoRoot, RepositoryData repo
...
         }
+        final Map<String, BlobContainer> indices= repository.getBlobContainer().children().get("indices").children();
+        // Assert that for each snapshot, the relevant metadata was written to index and shard folders
+        for (SnapshotId snapshotId: snapshotIds) {
+            final SnapshotInfo snapshotInfo = repository.getSnapshotInfo(snapshotId);
+            for (String index : snapshotInfo.indices()) {
+                final IndexId indexId = repositoryData.resolveIndexId(index);
+                assertThat(indices, hasKey(indexId.getId()));
+                final BlobContainer indexContainer = indices.get(indexId.getId());
+                assertThat(indexContainer.listBlobs(), hasKey("meta-" + snapshotId.getUUID() + ".dat"));
```
Comment (tlrx) [inline test/framework/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java]: nit: maybe reuse METADATA_NAME_FORMAT?

File: test/framework/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java [hunk 2]
```diff
@@ -113,14 +116,32 @@ private static void assertIndexUUIDs(BlobContainer repoRoot, RepositoryData repo
...
+                for (Map.Entry<String, BlobContainer> entry : indexContainer.children().entrySet()) {
+                    if (snapshotInfo.shardFailures().stream().noneMatch(shardFailure ->
+                        shardFailure.index().equals(index) != false && shardFailure.shardId() == Integer.parseInt(entry.getKey()))) {
+                        assertThat(entry.getValue().listBlobs(), hasKey("snap-" + snapshotId.getUUID() + ".dat"));
```
Comment (tlrx) [inline test/framework/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java]: nit: maybe reuse SNAPSHOT_NAME_FORMAT?

**Review by tlrx — APPROVED** [review]
LGTM, this is indeed a nice check to have. I left two minor comments

**Comment by original-brownbear** [comment]
@tlrx thanks! but ... appears this actually fails one of the cleanup tests ... looking into it :)

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2

**Comment by original-brownbear** [comment]
False alarm, just a missing exclusion on a test :) -> merging

**Pull request merged**